recipes/libpq/all/conanfile.py

```diff
@@ -156,12 +179,44 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "share"))
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
 
+    def _construct_library_name(self, name):
+        if self.settings.compiler == "Visual Studio":
+            return "lib{}".format(name)
+        return  name
+
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "PostgreSQL"
         self.cpp_info.names["cmake_find_package_multi"] = "PostgreSQL"
         self.env_info.PostgreSQL_ROOT = self.package_folder
-        self.cpp_info.libs = tools.collect_libs(self)
+        
+        self.cpp_info.components["pq"].libs = [self._construct_library_name("pq")]
+
+        if self.options.with_zlib:
+            self.cpp_info.components["pq"].requires.append("zlib::zlib")
+
+        if self.options.with_openssl:
+            self.cpp_info.components["pq"].requires.append("openssl::openssl")
+
+        if not self.options.shared:
+            if self.settings.compiler == "Visual Studio":
+                if tools.Version(self.version) < '12':
+                    self.cpp_info.components["pgport"].libs = ["libpgport"]
+                    self.cpp_info.components["pq"].requires.extend(["pgport"])
+                else:
+                    self.cpp_info.components["pgcommon"].libs = ["libpgcommon"]
+                    self.cpp_info.components["pgport"].libs = ["libpgport"]
+                    self.cpp_info.components["pq"].requires.extend(["pgport", "pgcommon"])
+            else:
+                if tools.Version(self.version) < '12':
+                    self.cpp_info.components["pgcommon"].libs = ["pgcommon"]
+                    self.cpp_info.components["pq"].requires.extend(["pgcommon"])
+                else:
+                    self.cpp_info.components["pgcommon"].libs = ["pgcommon", "pgcommon_shlib"]
+                    self.cpp_info.components["pgport"].libs = ["pgport", "pgport_shlib"]
+                    self.cpp_info.components["pq"].requires.extend(["pgport", "pgcommon"])
+
+
         if self.settings.os == "Linux":
-            self.cpp_info.system_libs = ["pthread"]
+            self.cpp_info.components["pq"].system_libs = ["pthread"]
         elif self.settings.os == "Windows":
-            self.cpp_info.system_libs = ["ws2_32", "secur32", "advapi32", "shell32", "crypt32", "wldap32"]
+            self.cpp_info.components["pq"].system_libs = ["ws2_32", "secur32", "advapi32", "shell32", "crypt32", "wldap32"]
```

Comment:
What is the cmake name of these components?

Reply:
* PostgreSQL::pq : The library itself
* PostgreSQL::pgcommon
* PostgreSQL::pgport